warehouse/query-core/src/main/java/datawave/query/tables/async/event/VisitorFunction.java

```diff
@@ -323,7 +332,14 @@ protected ASTJexlScript pushdownLargeFieldedLists(ShardQueryConfiguration config
                         
                         // sort from largest to smallest reductions and make reductions until under the threshold
                         Set<String> fieldsToReduce = new HashSet<>();
-                        int toReduce = termCount - config.getMaxTermThreshold();
+                        int toReduce;
+                        if (termCount > config.getMaxTermThreshold()) {
```

Comment:
if it hits both MaxTermThreshold & sourceLimit, does it matter that it only reduces down to MaxTermThreshold?